### PR TITLE
Imported mc_tool from mc py test suite; added py test for testing sc …

### PR DESCRIPTION
**Pull request by alsala**
…version handling

### CODE REVIEW

File: sdk/src/main/scala/com/horizen/block/WithdrawalEpochCertificate.scala [hunk 1]
```diff
@@ -10,21 +10,26 @@ import com.horizen.utils.{BytesUtils, Utils, VarInt}
...
 import com.horizen.librustsidechains.{Utils => ScCryptoUtils}
+import scorex.util.ScorexLogging
 
 import scala.util.Try
 
-case class FieldElementCertificateField(rawData: Array[Byte]) {
+case class FieldElementCertificateField(rawData: Array[Byte]) extends ScorexLogging {
   def fieldElementBytes(version: SidechainCreationVersion): Array[Byte] = {
-    version match {
+    logger.info("Fe before: " + BytesUtils.toHexString(rawData))
```
Comment (i-Alex) [inline sdk/src/main/scala/com/horizen/block/WithdrawalEpochCertificate.scala:19]: consider to remove logs later

File: qa/sc_versions_and_mc_certs.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+    def sc_setup_chain(self):
+        mc_node = self.nodes[0]
+        sc_node_configuration = SCNodeConfiguration(
+            MCConnectionInfo(address="ws://{0}:{1}".format(mc_node.hostname, websocket_port_by_mc_node_index(0)))
+        )
+        network = SCNetworkConfiguration(SCCreationInfo(mc_node, 100, self.sc_withdrawal_epoch_length, btr_data_length=0
+                                                        , sc_creation_version=1), sc_node_configuration)
+
+        mc_node.generate(450)
```
Comment (i-Alex) [inline qa/sc_versions_and_mc_certs.py:47]: do we need this? As far as I remember, `bootstrap_sidechain_nodes` will generate 450 blocks to switch to the proper fork with sidechain versions activation.

File: qa/sc_versions_and_mc_certs.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    def run_test(self):
+        time.sleep(0.1)
+        self.sync_all()
+        mc_node = self.nodes[0]
+        sc_node = self.sc_nodes[0]
+
+        print("Mc height = {}".format(mc_node.getblockcount()))
+        self.mcTest = CertTestUtils(self.options.tmpdir)
+
+        ret = create_alien_sidechain(self.mcTest, mc_node, scVersion=0, epochLength=9, customHexTag="5c00", feCfgList=[128, 128])
```
Comment (i-Alex) [inline qa/sc_versions_and_mc_certs.py:63]: let's use `epochLength=sc_withdrawal_epoch_length-1`.
Same below.

File: qa/sc_versions_and_mc_certs.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+Configuration:
+    Start 1 MC node and 1 SC node (with default websocket configuration).
+    SC node connected to the first MC node.
+
+Test:
+    TODO    
+"""
+
+
+class SCVersionsAndMCCertificates(SidechainTestFramework):
```
Comment (i-Alex) [inline qa/sc_versions_and_mc_certs.py:29]: Add test to `run_sc_tests`

File: qa/sc_versions_and_mc_certs.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    assert_true, create_alien_sidechain, create_certificate_for_alien_sc
+from SidechainTestFramework.sc_forging_util import *
+
+"""
+Configuration:
+    Start 1 MC node and 1 SC node (with default websocket configuration).
+    SC node connected to the first MC node.
+
+Test:
+    TODO
```
Comment (i-Alex) [inline qa/sc_versions_and_mc_certs.py:25]: need to add description.
Test itself seems fine.

**Review by i-Alex — CHANGES_REQUESTED** [review]

File: sdk/src/main/scala/com/horizen/block/WithdrawalEpochCertificate.scala (hunk 1 above)
Comment (alsala) [reply sdk/src/main/scala/com/horizen/block/WithdrawalEpochCertificate.scala]: I moved it to debug level

File: qa/sc_versions_and_mc_certs.py (hunk 2 above)
Comment (alsala) [reply qa/sc_versions_and_mc_certs.py]: You are right, this is redundant

File: qa/sc_versions_and_mc_certs.py (hunk 3 above)
Comment (alsala) [reply qa/sc_versions_and_mc_certs.py]: done

File: qa/sc_versions_and_mc_certs.py (hunk 4 above)
Comment (alsala) [reply qa/sc_versions_and_mc_certs.py:31]: Done

File: qa/sc_versions_and_mc_certs.py (hunk 5 above)
Comment (alsala) [reply qa/sc_versions_and_mc_certs.py]: Done

**Review by i-Alex — APPROVED** [review]

**Pull request merged**